storage/watch.go

```diff
@@ -0,0 +1,54 @@
+package storage
+
+import (
+	"sync"
+
+	"github.com/coreos/etcd/storage/storagepb"
+)
+
+// Watcher uses to watch on key-value store.
+type Watcher interface {
+	// Event returns a channel that receives observed event that matches the
+	// context of watcher. When watch finishes or is canceled or aborted, the
+	// channel is closed and returns empty event.
+	// Successive calls to Event return the same value.
+	Event() <-chan storagepb.Event
+
```

Comment:
ok. i am fine with keep this simple at first. but i hope we can keep this requirement in mind :)